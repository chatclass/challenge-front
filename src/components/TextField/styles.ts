import styled, { css, DefaultTheme } from 'styled-components'

import { TextFieldProps } from '.'

type IconPositionProps = Pick<TextFieldProps, 'iconPosition'>

type WrapperProps = Pick<TextFieldProps, 'disabled'> & { error?: boolean }

export const InputWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    border-radius: 0.2rem;
    padding: 0 1rem;
    border: 0.01rem solid;
    border-color: ${theme.colors.neutrals.gray[500]};
    background-color: white;
    &:focus-within {
      box-shadow: 0 0 0.2rem ${theme.colors.neutrals.black[400]};
    }
  `}
`

export const Input = styled.input<IconPositionProps>`
  ${({ theme, iconPosition }) => css`
    font-size: 1rem;
    color: ${theme.colors.neutrals.black[500]};
    padding: .75rem 0;
    padding-${iconPosition}: 1rem;
    border: 0;
    outline: none;
    width: ${iconPosition === 'right' ? `calc(100% - 2.2rem)` : `100%`};

    @media (min-width: ${({ theme }) => theme.breakpoints.md}) {
      font-size: 1.5rem;
      padding: 1.5rem 0;
    }
  `}
`

export const Icon = styled.div<IconPositionProps>`
  ${({ theme, iconPosition }) => css`
    display: flex;
    color: ${theme.colors.primary.orange};
    order: ${iconPosition === 'right' ? 1 : 0};
    & > svg {
      width: 2.2rem;
      height: 100%;
      color: ${theme.colors.neutrals.gray[600]};
    }
  `}
`

export const Error = styled.p`
  ${({ theme }) => css`
    margin: 1rem;
    color: ${theme.colors.neutrals.white[400]};
    font-size: 1rem;
  `}
`

const wrapperModifiers = {
  error: (theme: DefaultTheme) => css`
    ${InputWrapper} {
      border-color: ${theme.colors.primary.orange};
    }
    ${Icon} {
      color: ${theme.colors.primary.orange};
    }
  `,
  disabled: (theme: DefaultTheme) => css`
    ${Input},
    ${Icon} {
      cursor: not-allowed;
      color: ${theme.colors.primary.orange};
      &::placeholder {
        color: currentColor;
      }
    }
  `
}

export const Wrapper = styled.div<WrapperProps>`
  ${({ theme, error, disabled }) => css`
    ${error && wrapperModifiers.error(theme)}
    ${disabled && wrapperModifiers.disabled(theme)}
  `}
`
