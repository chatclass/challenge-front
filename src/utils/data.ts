/* eslint-disable @typescript-eslint/no-explicit-any */
const LIMIT = 6

export const normalizeData = (results: any) => {
  return {
    city: results.city,
    temperature: `${results.temp}°C ${results.description}`,
    min: `${
      results.forecast.find((f: any) => results.date.includes(f.date)).min
    }°`,
    max: `${
      results.forecast.find((f: any) => results.date.includes(f.date)).max
    }°`,
    wind: results.wind_speedy,
    humidity: `${results.humidity}%`,
    forecast: results.forecast
      .map((cast: any) => ({
        day: cast.weekday,
        min: `${cast.min}°`,
        max: `${cast.max}°`
      }))
      .filter((_cast: any, index: number) => index <= LIMIT)
  }
}
