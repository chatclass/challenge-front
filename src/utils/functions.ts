import { forecastDay } from '../store/ducks/weather/types';

export const unique = (
  value: forecastDay,
  index: number,
  self: forecastDay[],
) => {
  return (
    self
      .map(e => new Date(e.dt_txt).getDay())
      .indexOf(new Date(value.dt_txt).getDay()) === index
  );
};

export const capitalize = (str: string) => {
  const lower = str.toLowerCase();
  return str.charAt(0).toUpperCase() + lower.slice(1);
};
