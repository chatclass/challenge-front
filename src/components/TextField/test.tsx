import React from 'react'
import { render, screen } from '../../utils/test-utils'
import SearchIcon from '../../pages/Weather/searchIcon'

import TextField from '.'

describe('<TextField />', () => {
  it('Renders without Label', () => {
    render(<TextField />)

    expect(screen.queryByLabelText('Label')).not.toBeInTheDocument()
  })

  it('Renders with placeholder', () => {
    render(<TextField placeholder="Type your city here" />)

    expect(
      screen.getByPlaceholderText('Type your city here')
    ).toBeInTheDocument()
  })

  it('Renders with Icon', () => {
    render(<TextField icon={<SearchIcon data-testid="icon" />} />)

    expect(screen.getByTestId('icon')).toBeInTheDocument()
  })

  it('Renders with Icon on the right side', () => {
    render(
      <TextField
        icon={<SearchIcon data-testid="icon" />}
        iconPosition="right"
      />
    )

    expect(screen.getByTestId('icon').parentElement).toHaveStyle({ order: 1 })
  })

  it('Renders with error', () => {
    const { container } = render(
      <TextField
        icon={<SearchIcon data-testid="icon" />}
        error="Error message"
      />
    )

    expect(screen.getByText('Error message')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
