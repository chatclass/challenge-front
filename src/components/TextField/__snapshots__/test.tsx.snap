// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> Renders with error 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.2rem;
  padding: 0 1rem;
  border: 0.01rem solid;
  border-color: #777E90;
  background-color: white;
}

.c2:focus-within {
  box-shadow: 0 0 0.2rem #23262F;
}

.c5 {
  font-size: 1rem;
  color: #141416;
  padding: .75rem 0;
  padding-left: 1rem;
  border: 0;
  outline: none;
  width: 100%;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #ff7f00;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c4 > svg {
  width: 2.2rem;
  height: 100%;
  color: #353945;
}

.c6 {
  margin: 1rem;
  color: #FCFCFD;
  font-size: 1rem;
}

.c0 .c1 {
  border-color: #ff7f00;
}

.c0 .c3 {
  color: #ff7f00;
}

@media (min-width:768px) {
  .c5 {
    font-size: 1.5rem;
    padding: 1.5rem 0;
  }
}

<div
  class="c0"
>
  <div
    class="c1 c2"
  >
    <div
      class="c3 c4"
    >
      <svg
        data-testid="icon"
        fill="currentColor"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
    <input
      class="c5"
      type="text"
      value=""
    />
  </div>
  <p
    class="c6"
  >
    Error message
  </p>
</div>
`;
